client/src/views/client-area/ClientAreaHeader.vue

```diff
@@ -2,24 +2,139 @@
+      v-if="!isMobile() && !isSidebarMenuVisible()"
+      slot="icon-only"
+      id="trigger-toggle-menu-button"
+      class="ion-hide-lg-down topbar-button__item"
+      @click="resetSidebarMenu()"
+    >
+      <ion-icon
+        slot="icon-only"
+        :icon="menu"
+      />
+    </ion-button>
+    <ion-title class="header-title title-h1">{{ $msTranslate(title) }}</ion-title>
+
+    <div class="header-right">
+      <!-- log out -->
+      <ion-text
+        class="button-medium custom-button custom-button-danger-outline"
+        button
+        @click="logout"
+      >
+        <ion-icon :icon="logOut" />
+        {{ $msTranslate('clientArea.header.logout') }}
+      </ion-text>
+
+      <!-- preferences -->
+      <ion-text
+        class="button-medium custom-button"
+        button
+        @click="openPreferencesModal"
+      >
+        <ion-icon :icon="cog" />
+        {{ $msTranslate('clientArea.header.preferences') }}
+      </ion-text>
+
+      <!-- profile -->
+      <div
+        class="header-right-profile"
+        @click="goToPageClicked(ClientAreaPages.PersonalData)"
+      >
+        <user-avatar-name
+          :user-avatar="getUserName()"
+          :user-name="getUserName()"
+          :clickable="true"
+          class="avatar medium"
+        />
+      </div>
+    </div>
   </div>
 </template>
 
 <script setup lang="ts">
-import { IonButton } from '@ionic/vue';
-import { BmsAccessInstance } from '@/services/bms';
+import { BmsAccessInstance, PersonalInformationResultData } from '@/services/bms';
+import { IonText, IonTitle, IonIcon } from '@ionic/vue';
 import { navigateTo, Routes } from '@/router';
+import { Translatable } from 'megashark-lib';
+import { onMounted, ref } from 'vue';
+import { logOut, cog, menu } from 'ionicons/icons';
+import UserAvatarName from '@/components/users/UserAvatarName.vue';
+import { isMobile } from '@/parsec';
+import useSidebarMenu from '@/services/sidebarMenu';
+import { ClientAreaPages } from '@/views/client-area/types';
+
+const { isVisible: isSidebarMenuVisible, reset: resetSidebarMenu } = useSidebarMenu();
+
+defineProps<{
+  title: Translatable;
+}>();
+
+const personalInformation = ref<PersonalInformationResultData | null>(null);
+
+onMounted(async () => {
+  if (BmsAccessInstance.get().isLoggedIn()) {
+    personalInformation.value = await BmsAccessInstance.get().getPersonalInformation();
+  }
+});
 
 async function logout(): Promise<void> {
   await BmsAccessInstance.get().logout();
   await navigateTo(Routes.Home, { replace: true });
 }
+
+async function openPreferencesModal(): Promise<void> {
+  console.log('openPreferencesModal');
+}
+
+const emits = defineEmits<{
+  (e: 'pageSelected', page: ClientAreaPages): void;
+}>();
+
+async function goToPageClicked(page: ClientAreaPages): Promise<void> {
+  emits('pageSelected', page);
+}
+
+function getUserName(): string {
+  const info = personalInformation.value;
+  return `${info?.firstName} ${info?.lastName}` ?? '';
```

Comment:
I don't think this will work as expected. It will either result in `undefined undefined` or in `<space>`, both of which are not evaluated to false.

Reply:
fixed